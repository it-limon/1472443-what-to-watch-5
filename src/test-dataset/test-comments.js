export const testComments = [
  {
    id: 1,
    user: {
      id: 1,
      name: `User 1`
    },
    rating: 8,
    comment: `Comment 1`,
    date: `1995-12-17T03:24:00`
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `User 2`
    },
    rating: 7,
    comment: `Comment 2`,
    date: `1996-12-17T03:24:00`
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `User 3`
    },
    rating: 5,
    comment: `Comment 3`,
    date: `1997-12-17T03:24:00`
  }
];