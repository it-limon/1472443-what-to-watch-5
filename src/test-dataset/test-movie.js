export const testMovie = {
  id: 1,
  name: `El Camino: A Breaking Bad Movie`,
  posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
  previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
  backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
  backgroundColor: `#000000`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
  rating: 7.2,
  scoresCount: 36346,
  director: `Vince Gilligan`,
  starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
  runTime: 122,
  genre: `Drama`,
  released: 2019,
  isFavorite: true
};
