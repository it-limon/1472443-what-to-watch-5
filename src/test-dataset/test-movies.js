export const testMovies = [
  {
    id: 1,
    name: `El Camino: A Breaking Bad Movie`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
    rating: 7.2,
    scoresCount: 36346,
    director: `Vince Gilligan`,
    starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
    runTime: 122,
    genre: `Drama`,
    released: 2019,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Dark Knight`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, 
                  Batman must accept one of the greatest psychological and physical tests of his ability 
                  to fight injustice.`,
    rating: 9,
    scoresCount: 2271245,
    director: `Christopher Nolan`,
    starring: [`Christian Bale`, `Heath Ledger`, `Aaron Eckhart`],
    runTime: 152,
    genre: `Crime`,
    released: 2008,
    isFavorite: true
  }
];