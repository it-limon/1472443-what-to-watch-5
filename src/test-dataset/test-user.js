export const testUser = {
  id: 1,
  email: `user@mail.ru`,
  name: `User Name`,
  avatarUrl: `https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png`
};