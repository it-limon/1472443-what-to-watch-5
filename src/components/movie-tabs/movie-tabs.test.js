import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import MovieTabs from "./movie-tabs";
import {Provider} from "react-redux";
import {testMovie} from "../../test-dataset/test-movie";
import {testComments} from "../../test-dataset/test-comments";

let mockStore = null;
let store = null;

const noop = () => {};

describe(`Render MovieTabs`, () => {
  
  it(`Render MovieTabs with first active tab`, () => {
    const tree = renderer.create(
      <MovieTabs 
        movie={testMovie}
        activeIndex={0}
        onChangeActiveIndex={noop}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieTabs with second active tab`, () => {
    const tree = renderer.create(
      <MovieTabs 
        movie={testMovie}
        activeIndex={1}
        onChangeActiveIndex={noop}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render MovieTabs with third active tab`, () => {
    mockStore = configureStore([]);
    store = mockStore({
      DATA: {
        comments: testComments
      }
    });

    const tree = renderer.create(
      <Provider store={store}>
        <MovieTabs
          movie={testMovie}
          activeIndex={2}
          onChangeActiveIndex={noop}
        />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
