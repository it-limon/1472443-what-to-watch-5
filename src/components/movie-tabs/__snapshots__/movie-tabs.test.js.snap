// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieTabs Render MovieTabs with first active tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      7.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        36346 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Fugitive Jesse Pinkman runs from his captors, the law, and his past.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: Vince Gilligan
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Aaron Paul, Jonathan Banks, Matt Jones
      </strong>
    </p>
  </div>,
]
`;

exports[`Render MovieTabs Render MovieTabs with second active tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Vince Gilligan
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
        >
          Aaron Paul, Jonathan Banks, Matt Jones
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          2h 2m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Drama
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2019
        </span>
      </p>
    </div>
  </div>,
]
`;

exports[`Render MovieTabs Render MovieTabs with third active tab 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Comment 1
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              User 1
            </cite>
            <time
              className="review__date"
              dateTime="1995-12-17T03:24:00"
            >
              December 17, 1995
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8.0
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Comment 2
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              User 2
            </cite>
            <time
              className="review__date"
              dateTime="1996-12-17T03:24:00"
            >
              December 17, 1996
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          7.0
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Comment 3
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              User 3
            </cite>
            <time
              className="review__date"
              dateTime="1997-12-17T03:24:00"
            >
              December 17, 1997
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          5.0
        </div>
      </div>
    </div>
  </div>,
]
`;
